test/spec/test-common.js

```diff
@@ -304,6 +304,35 @@ describe('tus', () => {
       expect(options.onAfterResponse).toHaveBeenCalled()
     })
 
+    it('should invoke the onUploadUrlAvailable callback', async () => {
```

Comment:
This test covers the callback for HEAD responses. Can you also test its correct functionality with POST responses? It does not have to be a separate test but can also be added to another test covering POST requests.

Reply:
I have decided to remove the entier extra test that I had added, and rather test that the callback is called in the two conditions that you mentioned, in two corresponding, already existing tests. However, as it seems that the callback is only called after `testStack.nextRequest()` resolves, the `expect` call must be done after it (which looks a bit strange, but is logical, as I verified explictly).